records/state/state_test.go

```diff
@@ -167,15 +192,35 @@ func state(st string) statusOpt {
 	return func(s *Status) { s.State = st }
 }
 
-func netinfo(ips ...string) statusOpt {
+func netinfos(netinfos ...NetworkInfo) statusOpt {
 	return func(s *Status) {
-		netinfos := &s.ContainerStatus.NetworkInfos
-		for _, ip := range ips {
-			*netinfos = append(*netinfos, NetworkInfo{IPAddress: ip})
-		}
+		s.ContainerStatus.NetworkInfos = append(s.ContainerStatus.NetworkInfos, netinfos...)
 	}
 }
 
+func netinfo(ips ...string) NetworkInfo {
+	netinfo := NetworkInfo{}
+	for _, ip := range ips {
+		netinfo.IPAddresses = append(netinfo.IPAddresses, IPAddress{ip})
+	}
+	return netinfo
+}
+
+// NetworkInfo using v0.25 syntax for storing a single IP.
+func oldnetinfo(ip string) NetworkInfo {
+	netinfo := NetworkInfo{}
+	netinfo.IPAddress = ip
+	return netinfo
+}
+
+// NetworkInfo using both 0.25 and 0.26 syntax for IPs.
+func oldnewnetinfo(oldip string, newip string) NetworkInfo {
```

Comment:
My motivation here was to check that we gracefully handle malformed input,
since either syntax is accepted, the server could provide both if there is
an error.

Seems that it doesn't cost much to verify that, but happy to remove it if
you prefer to keep this code simple.

On Wed, Nov 18, 2015 at 8:21 AM James DeFelice <notifications@github.com>
wrote:

> In records/state/state_test.go
> <https://github.com/mesosphere/mesos-dns/pull/355#discussion_r45221011>:
>
> > +	netinfo := NetworkInfo{}
> > +	for _, ip := range ips {
> > +		netinfo.IPAddresses = append(netinfo.IPAddresses, IPAddress{ip})
> > +	}
> > +	return netinfo
> > +}
> > +
> > +// NetworkInfo using v0.25 syntax for storing a single IP.
> > +func oldnetinfo(ip string) NetworkInfo {
> > +	netinfo := NetworkInfo{}
> > +	netinfo.IPAddress = ip
> > +	return netinfo
> > +}
> > +
> > +// NetworkInfo using both 0.25 and 0.26 syntax for IPs.
> > +func oldnewnetinfo(oldip string, newip string) NetworkInfo {
>
> why? does mesos actually generate this? i thought there was no deprecation
> cycle between ipaddress and ipaddresses as far as mesos was concerned --
> that it was a clean break.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/355/files#r45221011>.
>